site/zenodo_rdm/subcommunities/schema.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Zenodo-RDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+"""Subcommunities schema implementation."""
+
+from invenio_communities.subcommunities.services.schema import (
+    MinimalCommunitySchema,
+    SubcommunityRequestSchema,
+)
+from invenio_i18n import gettext as _
+from marshmallow import ValidationError, fields, post_load
+
+
+class ZenodoCommunitySchema(MinimalCommunitySchema):
+    """Zenodo community schema.
+
+    Includes a project field.
+    """
+
+    project = fields.String()
+
+    @post_load(pass_original=True)
+    def process_project(self, data, original, **kwargs):
+        """Validate the project field.
+
+        For 'eu' community, it is a required field.
+        """
+        project = original.get("project")
+        is_ec_community = self.context.get("community_slug") == "eu"
```

Comment:
I am not convinced by this at all ... 

Having to hard-code the EU community slug here feels wrong, but we don't have a way to customize this request for specific communities yet. 

Perhaps it could "look better" if we add this as a config?